### ISSUE TITLE
No option for providing custom proxy for signature verifier

### ISSUE THREAD

**Issue by sharathsabbani** [amzn/alexa-skills-kit-java#69]
Most of the enterprise applications would behind proxy. It would be nice if we have option to pass in own proxy for signature verifier.

**Comment by vanackej** [comment]
Actually proxy was working perfectly using standard java properties (http.proxyHost, http.proxyPort [...]). Now it is broken and there is no documentation about how to configure proxy using your custom Proxy mechanism. I would strongly suggest to rollback this change and stick to standard Java proxy environment settings.

**Comment by breedloj** [comment]
Hey Johann, thanks for reporting this. I did not realize that the original change would have the side affect of explicitly overriding a user's proxy settings if they were providing them via environment variables. PR #142 restores the original behavior and should fix this issue, while at the same time retaining the ability to explicitly provide a proxy to be used by the servlet if so desired.

**Comment by vanackej** [comment]
Hi Jonathan. Thanks for the quick update ! Do you know when this change will be released in the sdk ? It is currently blocking our customer demos, just to know if I should patch my code for now, or if I just wait for a new release to be done.

**Comment by breedloj** [comment]
2.5.4 is out now and contains this fix. :)

**Comment by vanackej** [comment]
Awesome !

**Comment by vanackej** [comment]
Well, one more issue with this : Out of the box, the proxy is still forced to NO_PROXY in SkillServlet [here](https://github.com/alexa/alexa-skills-kit-sdk-for-java/blob/2.0.x/ask-sdk-servlet-support/src/com/amazon/ask/servlet/SkillServlet.java#L69)
I really think the default behavior should comply with Java conventions for proxy (i.e. using standard env variables)

**Comment by breedloj** [comment]
Yep, you are totally correct. That's a miss on my end with the change. I'll make sure a follow up change gets pushed out shortly.

**Comment by breedloj** [comment]
Hey Johann, 2.5.5 contains the follow-up fix for this. Let me know if you have any other issues.
